client/consul_template.go

```diff
@@ -492,3 +504,60 @@ func runnerConfig(config *config.Config, vaultToken string) (*ctconf.Config, err
 	conf.Finalize()
 	return conf, nil
 }
+
+// loadTemplateEnv loads task environment variables from templates.
+func loadTemplateEnv(builder *env.Builder, taskDir string, t *structs.Template) error {
+	if !t.Envvars {
+		return nil
+	}
+	f, err := os.Open(filepath.Join(taskDir, t.DestPath))
+	if err != nil {
+		return fmt.Errorf("error opening env template: %v", err)
+	}
+	defer f.Close()
+
+	// Parse environment fil
+	vars, err := parseEnvFile(f)
+	if err != nil {
+		return fmt.Errorf("error parsing env template %q: %v", t.DestPath, err)
+	}
+
+	// Set the environment variables
+	builder.SetTemplateEnv(vars)
+	return nil
+}
+
+// parseEnvFile and return a map of the environment variables suitable for
+// TaskEnvironment.AppendEnvvars or an error.
+//
+// See nomad/structs#Template.Envvars comment for format.
+func parseEnvFile(r io.Reader) (map[string]string, error) {
```

Comment:
Since this is tied to a template it must be a single file. It's probably worth documenting how to use env files from artifacts by setting source path to where the artifact is downloaded, but env dirs (or autoloading `.env` files) is out of scope for this feature.